Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -15,9 +15,14 @@ import POSIX
 //FIXME: Incremental builds
 
 extension Command {
-    static func compile(clangModule module: ClangModule, externalModules: Set<Module>, configuration conf: Configuration, prefix: String) -> (Command, Command) {
-
-        let wd = Path.join(prefix, "\(module.c99name).build")
+    static func compile(clangModule module: ClangModule, externalModules: Set<Module>, configuration conf: Configuration, prefix: String) throws -> (Command, Command) {
+        
+        let wd = module.workingDirectory(prefix)
+        
+        if module.type == .Library {
+            try module.generateModuleMap(inDir: wd)
```

Comment:
For now I can't think of a direct way to do that because we try to generate module map when building C targets and module maps are needed for swift modules but I get your point.